src/ArrayContainer.php

```diff
@@ -32,6 +38,7 @@ public function __construct($data = array())
         } else {
             throw new \InvalidArgumentException('The ArrayContainer requires either an array or an array-like object');
         }
+        $this->delegateLookupContainer = $delegateLookupContainer ?: $this;
```

Comment:
Replacing the delegate container with `new FailoverOnMissContainer($delegateContainer, $this)` is doing/overriding the delegate container's job. If you do that here then there is no reason it shouldn't be done for every other implementation, and that defeats the specification of the "delegate lookup feature".

The goal is to delegate the lookup to something else, so we shouldn't make assumptions (i.e. preemptively fix it) on what that delegate does.